traits/src/signature/arrayref.rs

```diff
@@ -0,0 +1,92 @@
+//! This module contains the traits and related errors for signing and verification where arguments
+//! are provided as array references, and outputs are written to mutable array references.
+
+use libcrux_secrets::U8;
+
+/// A signer. This trait is the most low-level and mostly used in the implementation of other, more
+/// usable APIs on top. This trait takes array references as arguments.
+///
+/// The `SignAux` type is auxiliary information required for signing.
+pub trait Sign<const SIGNING_KEY_LEN: usize, const SIGNATURE_LEN: usize> {
+    /// Auxiliary information needed for signing.
+    type SignAux<'a>;
+    /// Sign a payload using a provided signature key. Required auxiliary information is provided using
+    /// the `aux` argument.
+    fn sign(
+        payload: &[u8],
+        signing_key: &[U8; SIGNING_KEY_LEN],
+        signature: &mut [u8; SIGNATURE_LEN],
+        aux: Self::SignAux<'_>,
+    ) -> Result<(), SignError>;
+}
+
+/// A verifier. This trait takes array references as arguments.
+///
+/// The `VerifyAux` type is auxiliary information required for verification.
+pub trait Verify<const VERIFICATION_KEY_LEN: usize, const SIGNATURE_LEN: usize> {
+    /// Auxiliary information needed for verification.
+    type VerifyAux<'a>;
+    /// Verify a signature using a provided verification key. Required auxiliary information is provided using
+    /// the `aux` argument.
+    fn verify(
+        payload: &[u8],
+        verification_key: &[u8; VERIFICATION_KEY_LEN],
+        signature: &[u8; SIGNATURE_LEN],
+        aux: Self::VerifyAux<'_>,
+    ) -> Result<(), VerifyError>;
+}
```

Comment:
Why did you decide to split this into two traits? I think this is a valid way to design the traits, but it deviates from the traits we already have.